pkg/scheduler/scheduler.go

```diff
@@ -252,7 +254,11 @@ func (sched *Scheduler) Run() {
 		return
 	}
 
-	go wait.Until(sched.scheduleOne, 0, sched.config.StopEverything)
+	go func() {
+		sched.mu.Lock()
```

Comment:
The `Run` function is not called by multiple threads. Why do we need synchronization here?